pyccel/ast/numpyext.py

```diff
@@ -819,30 +819,47 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
-    __slots__ = ('_shape','_rank','_order')
+    __slots__ = ('_shape','_rank','_order','_arg','_precision')
     _dtype = NativeReal()
-    _precision = default_precision['real']
 
-    def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
-        if self.dim is not None:
+    def __init__(self, arg, axis=None):
+        super().__init__(arg, axis)
+        cmplx_precision = None
+        if isinstance(arg.dtype, NativeBool):
+            arg = PythonFloat(arg)
+        elif not isinstance(arg.dtype, (NativeComplex, NativeReal)):
+            arg = PythonFloat(arg)
+        self._arg = PythonList(arg) if arg.rank == 0 else arg
+        self._precision = cmplx_precision if cmplx_precision else arg.precision
+        if self.axis is not None:
             sh = list(arg.shape)
-            del sh[self.dim]
+            del sh[self.axis]
             self._shape = tuple(sh)
             self._order = arg.order
         else:
             self._shape = ()
             self._order = None
         self._rank = len(self._shape)
-        super().__init__(arg, dim)
+        self._order = arg.order
```

Comment:
I can't understand what is happening with the variable `cmplx_precision`.

It is set to `None` on line 827 and then its value is checked on line 833... but we know that the value is `None`, of course!

Can you explain?

Reply:
i was using `PythonAbs` in case of `arg` is a complex type, and because we to save the precision of `arg` and `PythonAbs` has a default precision which is 8, when passing complex arg with 4 precision, `PythonAbs` will change it to 8 precision which we didn't want to have, so i had to store the original precision of before passing `arg` to `PythonAbs`. but now we are handling complex types in the Fortran printer since Fortran `Norm2` has a problem with complex type and Python not, so this variable is useless.

Reply:
Thanks for removing it.